examples/notify.php

```diff
@@ -12,13 +12,53 @@
 #[AsTask(description: 'Sends a notification when the task finishes')]
 function notify_on_finish(): void
 {
-    run(['sleep', '1'], notify: true);
+    run(['sleep', '0.1'], notify: true);
 }
 
 #[AsTask(description: 'Sends a notification')]
 function send_notify(): void
 {
-    notify('Hello world!');
+    // The two tasks below will don't send a notification
+    with(
+        callback: function () {
+            run(['sleep', '0.1']); // Will not send a notification
+            notify('Hello world! (not send)'); // Will not send a notification
+
+            run(['sleep', '0.1'], notify: true); // Will send a notification
+        },
+        context: context()->withNotify(false)
+    );
+
+    // This will send a notification
+    with(
+        callback: function () {
+            run(['sleep', '0.1']); // Will send a notification
+            notify('Hello world! (send with true)'); // Will send a notification
+
+            run(['sleep', '0.1'], notify: false); // Will not send a notification for this specific run
+        },
+        context: context()->withNotify(true)
+    );
+
+    // This will send a notification partially
```

Comment:
```suggestion
    // This will send a notification partially, this is the default behavior
```

➕ could you move this code block at the beginning of the file?